packages/common/src/ui/components/buttons/Button.tsx

```diff
@@ -1,30 +1,59 @@
 import React from 'react';
-import { Button as MuiButton } from '@material-ui/core';
+import { Button as MuiButton, Tooltip } from '@material-ui/core';
 import { styled } from '@material-ui/core/styles';
 import { LocaleKey, useTranslation } from '../../../intl/intlHelpers';
 import { DefaultButtonStyles } from './styles';
+import { useIsSmallScreen } from '../../../hooks/useIsSmallScreen';
 
 interface ButtonProps {
   icon: React.ReactNode;
   labelKey: LocaleKey;
   onClick: (event: React.MouseEvent<HTMLButtonElement>) => void;
+  shouldShrink?: boolean;
```

Comment:
I'm surprised- I thought you generally favoured specific versions of components rather than conditionals?
I hate to say it - but I really like the idea of a `ShrinkableButton` 

I was thinking it might end up a switch between an `IconButton` and a `Button` - instead you've done wonders! :smiley:  the transition on the width is fantastic

Reply:
Hmm yeah I liked the idea of it also and I actually wrote it similar to this as it's own component but it was just so minimally different I couldn't justify it 🤷 

I'd be happy to change if you think it's a good idea

Reply:
I thought there were enough differences between the two - you would have a better idea than I would though, so I'd say stick with what you have.